private_sharing/views.py

```diff
@@ -268,9 +270,18 @@ class AuthorizeOAuth2ProjectView(ConnectedSourcesMixin, ProjectMemberMixin,
     template_name = 'private_sharing/authorize-oauth2.html'
 
     def dispatch(self, *args, **kwargs):
+        client_id = self.request.GET.get('client_id', None)
```

Comment:
Is `self.get_object` not going to work here? Because this seems redundant to that...

Reply:
sorry, I guess the thing to look at is the self.application property https://github.com/OpenHumans/open-humans/blob/master/common/views.py#L35 ...this code seems unnecessary (see below)...